tests/integration/servicemesh/smmr_test.go

```diff
@@ -110,33 +151,55 @@ func verifyThatIngressHasVirtualHostForMember(ctx framework.TestContext, expecte
 	}, retry.Timeout(10*time.Second))
 }
 
-func verifyThatRouteExistsOrFail(ctx framework.TestContext, expectedGwNs, expectedGwName, expectedHost string) {
-	routerClient, err := ior.NewRouterClient()
-	if err != nil {
-		ctx.Fatalf("failed to create Router client: %s", err)
-	}
-	var routes *routeapiv1.RouteList
+func verifyThatRouteExistsOrFail(ctx framework.TestContext, gatewayNamespace, gatewayName, host string) {
+	routeClient := ctx.AllClusters().Default().Route()
+
 	retry.UntilSuccessOrFail(ctx, func() error {
-		routes, err = routerClient.Routes("istio-system").List(context.TODO(), metav1.ListOptions{})
+		route, err := findRoute(routeClient, gatewayNamespace, gatewayName, host)
 		if err != nil {
-			return fmt.Errorf("failed to get Routes: %s", err)
+			return fmt.Errorf("failed to get Routes: %v", err)
 		}
-		if len(routes.Items) == 0 {
-			return fmt.Errorf("no Routes found")
+
+		if route == nil {
+			return fmt.Errorf("no Route found")
 		}
+
 		return nil
-	}, retry.Timeout(10*time.Second))
+	}, retry.Timeout(300*time.Second))
```

Comment:
I don't think such a high timeout is necessary, since I've seen the test fail even with these 300s. 

I think these failed tests show that some kind of race condition is still present (either in the production code or the tests themselves).

Reply:
> I don't think such a high timeout is necessary, since I've seen the test fail even with these 300s.
Yes that is what I have found out with experimenting the increase of this timeout. The tests will still fail even with a long timeout. I will change them all back.